modules/@apostrophecms/piece-type/ui/apos/components/AposImageCropper.vue

```diff
@@ -59,42 +83,268 @@ export default {
+      // In that case, use regular elem.style.transform and extract values with a regex
+      const matrix = new window.DOMMatrixReadOnly(stencilStyle.transform);
+
+      this.stencilCoordinates = {
+        left: matrix.m41,
+        top: matrix.m42,
+        width: stencilElement.clientWidth,
+        height: stencilElement.clientHeight
+      };
+    },
+    /**
+     * Return the size and half size of the focal point element,
+     * used for its DOM manipulation and relative position calculation.
+     */
+    getFocalPointSize () {
+      const { focalPoint } = this.$refs;
+
+      const size = {
+        width: focalPoint.clientWidth + focalPoint.clientLeft * 2,
+        height: focalPoint.clientHeight + focalPoint.clientTop * 2
+      };
+
+      return {
+        ...size,
+        halfWidth: size.width / 2,
+        halfHeight: size.height / 2
+      };
+    },
+    /**
+     * Place the focal point at its current coordinates
+     * inside the stencil or at the center of it by default.
+     */
+    placeFocalPointInStencil () {
+      const { focalPoint } = this.$refs;
+
+      const x = this.docFields.data.x || 50;
+      const y = this.docFields.data.y || 50;
+
+      const focalPointSize = this.getFocalPointSize();
+
+      const left = Math.round(x / 100 * this.stencilCoordinates.width + this.stencilCoordinates.left - focalPointSize.halfWidth);
+      const top = Math.round(y / 100 * this.stencilCoordinates.height + this.stencilCoordinates.top - focalPointSize.halfHeight);
+
+      focalPoint.style.left = `${left}px`;
+      focalPoint.style.top = `${top}px`;
+    },
+    /**
+     * Get focal point relative position inside the stencil
+     * and emit it to update `x` and `y` as percentages,
+     * or as `null` if outside it.
+     */
+    updateFocalPointCoordinates () {
+      const { focalPoint } = this.$refs;
+
+      const focalPointSize = this.getFocalPointSize();
+
+      const x = (focalPoint.offsetLeft + focalPointSize.halfWidth - this.stencilCoordinates.left) / this.stencilCoordinates.width;
+      const y = (focalPoint.offsetTop + focalPointSize.halfHeight - this.stencilCoordinates.top) / this.stencilCoordinates.height;
+
+      const sanitizeCoordinates = ({ x, y }) => (x >= 0 && x <= 1 && y >= 0 && y <= 1) ? {
+        x: Math.abs(Math.round(x * 100)),
+        y: Math.abs(Math.round(y * 100))
+      } : {
+        x: null,
+        y: null
+      };
+
+      const coordinates = sanitizeCoordinates({
+        x,
+        y
+      });
+
+      this.$emit('change', coordinates, false);
     }
   }
 };
 </script>
 <style lang='scss'>
 .apos-image-cropper {
+  position: relative;
   max-width: 100%;
+  cursor: pointer;
+
+  .apos-image-focal-point {
+    position: absolute;
+    z-index: 1;
+    width: 10px;
+    height: 10px;
+    border-radius: 50%;
+    border: 1px solid var(--a-white);
+    background-color: var(--a-primary);
+    box-shadow: 0 0 4px var(--a-black);
+    transition: left 0.1s ease, top 0.1s ease;
```

Comment:
_ℹ️  transition is used when focal point is being re-positioned after a screen resize, to make it a bit smoother._